### PR TITLE
Rework the test package tester to avoid grep pitfalls.

### PR DESCRIPTION
**Pull request by ncabatoff**
I observed test-generate-test-package-lists.sh fail on a backport that didn't fail on main, and it wasn't because the test package it complained about was already assigned to a test package on main.  I believe the bug stems from the use of grep with `\b`, which is a word boundary.  The intent was to have it treat that as a whitespace boundary delimiter, but it's actually a letter boundary delimiter.

I was uncomfortable with leveraging grep for this purpose, probably because I never got really comfortable with bash arrays.  Either way, my fix probably goes further than strictly necessary to fix the bug, but I lack the facility with this aspect of bash to make a more targeted fix. To me the result is simpler easier to understand so I'm ok with that.

### CODE REVIEW

File: .github/scripts/test-generate-test-package-lists.sh [hunk 1]
```diff
@@ -49,18 +37,24 @@ count_test_packages() {
...
-all_packages=()
-
-cd "$dir/../.."
-get_module_packages
-
-cd "$dir/../../sdk"
-get_module_packages
+contains() {
+    target=$1; shift
+    for i; do
```
Comment (raskchanky) [inline .github/scripts/test-generate-test-package-lists.sh:42]: What is this looping over?

**Review by VioletHynes — APPROVED** [review]
LGTM!

File: .github/scripts/test-generate-test-package-lists.sh (hunk 1 above)
Comment (raskchanky) [reply .github/scripts/test-generate-test-package-lists.sh:42]: Ah, I didn't realize that was implicit

**Review by raskchanky — APPROVED** [review]

**Comment by ncabatoff** [comment]
Manual 1.11 backport: https://github.com/hashicorp/vault/pull/21289

**Comment by ncabatoff** [comment]
Manual 1.12 backport: https://github.com/hashicorp/vault/pull/21290

**Pull request merged**